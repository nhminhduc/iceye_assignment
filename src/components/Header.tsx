import { isDarkModeAtom } from "@/store/uiAtoms";
import { Layout, Switch } from "antd";
import { useAtom } from "jotai";

const { Header: AntdHeader } = Layout;

const Header = () => {
  const [isDarkMode, setIsDarkMode] = useAtom(isDarkModeAtom);

  return (
    <AntdHeader
      className="flex items-center justify-end px-4 shadow-sm"
      role="banner"
    >
      <Switch
        checkedChildren="🌙"
        unCheckedChildren="☀️"
        checked={isDarkMode}
        onChange={(checked) => setIsDarkMode(checked)}
        aria-label="Toggle dark/light mode"
      />
    </AntdHeader>
  );
};

export default Header;
