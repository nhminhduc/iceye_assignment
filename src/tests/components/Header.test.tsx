import { render, screen } from "@testing-library/react";
import { Provider } from "jotai";
import { describe, it, expect } from "vitest";
import Header from "@/components/Header";

describe("Header component", () => {
  it("renders the header with a switch snapshots", () => {
    const { container } = render(
      <Provider>
        <Header />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it("renders the header with a switch", () => {
    render(
      <Provider>
        <Header />
      </Provider>
    );

    // Check if the switch/toggle is present
    const switchElement = screen.getByRole("switch", {
      name: /toggle dark\/light mode/i,
    });
    expect(switchElement).toBeInTheDocument();
  });
});
