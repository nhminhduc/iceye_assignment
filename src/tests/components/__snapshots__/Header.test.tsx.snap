// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header component > renders the header with a switch snapshots 1`] = `
<div>
  <header
    class="ant-layout-header flex items-center justify-end px-4 shadow-sm css-dev-only-do-not-override-1m63z2v"
    role="banner"
  >
    <button
      aria-checked="false"
      aria-label="Toggle dark/light mode"
      class="ant-switch css-dev-only-do-not-override-1m63z2v"
      role="switch"
      type="button"
    >
      <div
        class="ant-switch-handle"
      />
      <span
        class="ant-switch-inner"
      >
        <span
          class="ant-switch-inner-checked"
        >
          🌙
        </span>
        <span
          class="ant-switch-inner-unchecked"
        >
          ☀️
        </span>
      </span>
    </button>
  </header>
</div>
`;
